soc/nordic/nrf54h/gpd/gpd.c

```diff
@@ -181,11 +208,25 @@ int nrf_gpd_request(uint8_t id)
 		return -EIO;
 	}
 
-	sys_notify_init_spinwait(&client.notify);
+	if (k_is_pre_kernel()) {
+		sys_notify_init_spinwait(&client.notify);
+	} else {
+		sys_notify_init_callback(&client.notify, request_cb);
+		k_mutex_lock(&gpd_mgr->lock, K_FOREVER);
+	}
 
-	onoff_request(&gpd_mgr->mgr, &client);
+	ret = onoff_request(&gpd_mgr->mgr, &client);
+	if (ret < 0) {
+		return ret;
```

Comment:
Failure leaves the mutex locked in the pre_kernel flow

Reply:
good catch